File: java/creational_pattern/prototype/Circle.java
package prototype;

public class Circle implements Shape {
    private String color;

    public Circle(String color) {
        this.color = color;
    }

    @Override
    public Shape clone() {
        return new Circle(this.color);
    }

    public void setColor(String color) {
        this.color = color;
    }
}

File: typescript/creational_pattern/src/protoype/Circle.ts
import Shape from "./Shape";

class Circle implements Shape {
    private color: string;

    public constructor(color: string) {
        this.color = color;
    }

    public clone(): Shape {
        return new Circle(this.color);
    }

    public set setColor(color: string) {
        this.color = color;
    }
}

export default Circle;